src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalCatchCheck.java

```diff
@@ -129,17 +129,13 @@ public void visitToken(DetailAST detailAST) {
         if (currentNode.getType() == TokenTypes.BOR) {
             exceptionTypes.addAll(getAllExceptionTypes(currentNode));
             currentNode = currentNode.getNextSibling();
-            if (currentNode != null) {
```

Comment:
`exceptionTypes` is only sent to `FullIdent.createFullIdent` which is `null` safe, so it doesn't matter if we send it `null` or not.